lib/widgets/lightbox.dart

```diff
@@ -232,19 +252,54 @@ class _ImageLightboxPageState extends State<_ImageLightboxPage> {
     );
   }
 
+  Widget _frameBuilder(BuildContext context, Widget child, int? frame, bool wasSynchronouslyLoaded) {
+    if (widget.thumbnailUrl == null) return child;
+
+    // The full image is available, so display it.
+    if (frame != null) return child;
+
+    // Display the thumbnail image while original image is downloading.
+    return RealmContentNetworkImage(widget.thumbnailUrl!,
+      filterQuality: FilterQuality.medium);
+  }
+
+  Widget _loadingBuilder(BuildContext context, Widget child, ImageChunkEvent? loadingProgress) {
+    if (widget.thumbnailUrl == null) return child;
+
+    // `loadingProgress` becomes null when Image has finished downloading.
+    final double? progress = loadingProgress?.expectedTotalBytes == null ? null
+      : loadingProgress!.cumulativeBytesLoaded / loadingProgress.expectedTotalBytes!;
+
+    if (progress != _loadingProgress) {
+      _loadingProgress = progress;
+      // This function is called in a build method and setState
+      // can't be called in a build method, so delay it.
+      SchedulerBinding.instance.scheduleFrameCallback((_) { if (mounted) setState(() {}); });
```

Comment:
nit: lots going on on this line (lots of control flow, with two function literals and a conditional); split it up

In fact I think it benefits from being expanded slightly more, to turn the if-mounted check into an early return to match the pattern it usually appears as:

```suggestion
      SchedulerBinding.instance.scheduleFrameCallback((_) {
        if (!mounted) return;
        setState(() {});
      });
```